src/python/pants/backend/java/target_types.py

```diff
@@ -142,6 +144,40 @@ async def generate_targets_from_java_sources(
     )
 
 
+# Things for JARs
+#
+
+
+class JvmMainClassName(StringField):
+    alias = "main"
+    required = True
+    help = (
+        "`.`-separated name of the JVM class containing the `main()` method to be called when "
+        "executing this JAR."
+    )
+
+
+class JvmRootClassAddress(StringField):
+    alias = "root_address"
+    required = True
+    help = "Address that contains the Java source which contains the `main` class."
+
+
+class FatJar(Target):
```

Comment:
Is there a better name than "fat jar"? I've sometimes seen these called "deploy jars."

Reply:
cc @stuhood @Eric-Arellano for bike shedding

Reply:
Yea, `deploy` jar is fairly widely used, and better describes why you build one of these (it's definitely not for publishing).

Reply:
I'd suggest using `deploy` jar, but referencing the other name `'aka "fat" jars'` in the help string.

Reply:
Makes sense! I'll make those changes.

Reply:
I like "deploy" jar much more, good idea.